src/main/java/Duke.java

```diff
@@ -84,6 +137,32 @@ public static void addList() {
         System.out.println("    ____________________________________________________________");
     }
 
+    private static void markAsDone(int index) {
+        System.out.println("    ____________________________________________________________");
+        System.out.println("     Nice! I've marked this task as done:");
+        String description = list.get(index - 1);
+        String time = date.get(index - 1);
+        String category = type.get(index - 1);
+        // mark as done
+        done.set(index - 1, 1);
+        switch (category) {
+        case "T":
+            Todo todo = new Todo(description);
+            System.out.println("       " + todo.doneToString());
+            break;
+        case "D":
+            Deadline deadline = new Deadline(description, time);
+            System.out.println("       " + deadline.doneToString());
+            break;
+        case "E":
+            Event event = new Event(description, time);
+            System.out.println("       " + event.doneToString());
+            break;
```

Comment:
A similar switch statement is found in the main() method. Good to abstract and avoid redundant coding.